packages/flutter_tools/lib/src/flutter_project_metadata.dart

```diff
@@ -311,11 +351,58 @@ migration:
       }
     }
   }
+
+  /// Performs a biased 3-way merge on current, base, and target MigrateConfigs.
+  ///
+  /// The base is the common ancestor config, while the target is a newly generated
+  /// config modern config. Current is the existing config in the project. 
+  ///
+  /// This merge is biased such that the results are consistent with the
+  /// way project migration occurs such as not updating create_revision.
+  factory MigrateConfig merge(MigrateConfig current, MigrateConfig base, MigrateConfig target, Logger logger) {
```

Comment:
I don't think this compiles?
```suggestion
  factory MigrateConfig.merge(MigrateConfig current, MigrateConfig base, MigrateConfig target, Logger logger) {
```